### DISCUSSION TITLE
pyproject.toml support (As project file and file format)

### DISCUSSION THREAD

**Discussion by djongepier** [pyscripter/pyscripter#1338] [Ideas]
Hi all,

Having used PyScripter as a nice lightweight IDE for python, some things could be improved.
Most notably, I think it would be a good idea to embrace the pyproject.toml format.

1. One way is to support the TOML format. This way the editor will not blind me when I open the file for editing.

2. Another way, but that's really throwing it out there, support the pyproject.toml as THE format for projects in PyScripter.

This is just a suggestion, I do understand that the scope of both options is wildly different. Unfortunately, I'm not able to help since I can barely do Python, and no Delphi/Pascal.

Thank you for this editor, and thanks for reading.

Greetings,

Daniël

**Comment by pyscripter** [comment]
Actually the project format is "INI" files which is very similar to TOML. (see this [discussion](https://github.com/toml-lang/toml/discussions/845)).

> 1. This way the editor will not blind me when I open the file for editing.

Could you please explain?

**Comment by djongepier** [comment]
See this screenshot, when opening the TOML file, it's not with a dark background like the py files.

![image](https://github.com/user-attachments/assets/b535891d-bcd6-475f-97fe-8fffefabb332)

For the pyproject.toml thing I mentioned, seeing that the project file for PyScripter also stores what files belong to the project, I presume it's not possible to say, ditch the current project format and use the pyproject.toml for Pyscripter's projects too.

**Comment by pyscripter** [comment]
I see.  you want syntax highlighting for toml files matching the color scheme of other highlighters.  Good idea.  Will do.

**Comment by pyscripter** [comment]
Actually, you can get pretty good results by doing View, Editor, Syntax, INI.
But the support for toml files will be perfect in the next version!
